### PR TITLE
support server async responses

### PR DESCRIPTION
**Pull request by mehdijouan**
Hi,

Here is a PR to support asynchronous requests from a raml (see #104 ).
I have modified rules to support a configuration named `callableResponse`.
Generated code will look like this [blog post](https://spring.io/blog/2012/05/10/spring-mvc-3-2-preview-making-a-controller-method-asynchronous/). I preferred to use `Callable` instead of `DeferredResult` as a first step.

* `DeferredResult` handling will come in another PR.
* Client side code will come in another PR.

I hope it is acceptable.

### CODE REVIEW

File: springmvc-raml-parser/src/main/java/com/phoenixnap/oss/ramlapisync/generation/rules/spring/SpringCallableResponseEntityRule.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * endpoint declares a response body the first type of the response body will added as a generic
+ * type to the ResponseEntity.
+ *
+ * #%RAML 0.8 title: myapi mediaType: application/json baseUri: /
+ *
+ * /base: get: /{id}: get: responses: 200: body: application/json: schema: NamedResponseType ...
+ *
+ * OUTPUT: {@literal @}Callable{@literal <}ResponseEntity{@literal <}NamedResponseType{@literal >}{@literal >}
+ *
+ * @author armin.weisser
```
Comment (stojsavljevic) [inline springmvc-raml-parser/src/main/java/com/phoenixnap/oss/ramlapisync/generation/rules/spring/SpringCallableResponseEntityRule.java]: Can you please put appropriate @author here and in all other classes?

File: springmvc-raml-parser/src/main/java/com/phoenixnap/oss/ramlapisync/generation/rules/spring/SpringControllerDecoratorRule.java [hunk 2]
```diff
@@ -63,7 +68,18 @@
  * @author armin.weisser
  * @since 0.4.1
  */
-public abstract class SpringControllerDecoratorRule implements Rule<JCodeModel, JDefinedClass, ApiResourceMetadata> {
+public abstract class SpringControllerDecoratorRule implements ConfigurableRule<JCodeModel, JDefinedClass, ApiResourceMetadata> {
+    public static final String CALLABLE_RESPONSE_CONFIGURATION = "callableResponse";
+
+    private boolean callableResponse;
```
Comment (stojsavljevic) [inline springmvc-raml-parser/src/main/java/com/phoenixnap/oss/ramlapisync/generation/rules/spring/SpringControllerDecoratorRule.java]: Can you please initialize configuration variables?
Even though they default to false, it's not super readable this way.

File: springmvc-raml-parser/src/main/java/com/phoenixnap/oss/ramlapisync/generation/rules/spring/SpringCallableResponseEntityRule.java (hunk 1 above)
Comment (mehdijouan) [inline springmvc-raml-parser/src/main/java/com/phoenixnap/oss/ramlapisync/generation/rules/spring/SpringCallableResponseEntityRule.java]: Done

File: springmvc-raml-parser/src/main/java/com/phoenixnap/oss/ramlapisync/generation/rules/spring/SpringControllerDecoratorRule.java (hunk 2 above)
Comment (mehdijouan) [inline springmvc-raml-parser/src/main/java/com/phoenixnap/oss/ramlapisync/generation/rules/spring/SpringControllerDecoratorRule.java]: done

**Pull request merged**